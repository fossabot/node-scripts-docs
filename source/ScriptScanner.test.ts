import { ScriptScanner } from "./ScriptScanner";

it("finds existing manifests (simple project)", async () => {
  const scanner = new ScriptScanner("test/fixtures/default");
  const manifests = await scanner.findManifests();
  expect(manifests).toEqual(["package.json"]);
});

it("finds existing manifests (monorepo)", async () => {
  const scanner = new ScriptScanner("test/fixtures/monorepo");
  const manifests = await scanner.findManifests();
  expect(manifests).toEqual(
    expect.arrayContaining(["package.json", "packages/some-project/package.json"])
  );
});

it("loads existing manifests (simple project)", async () => {
  const scanner = new ScriptScanner("test/fixtures/default");
  await scanner.loadManifests();
  const store = await scanner.loadScripts();
  expect(store.scripts).toMatchSnapshot();
});

it("loads existing manifests (monorepo)", async () => {
  const scanner = new ScriptScanner("test/fixtures/monorepo");
  await scanner.loadManifests();
  const store = await scanner.loadScripts();
  expect(store.scripts).toMatchSnapshot();
});

it("signals file can be loaded", async () => {
  expect(await ScriptScanner.canLoad("test/fixtures/default/package.json")).toBe(true);
  expect(await ScriptScanner.canLoad("invalid")).toBe(false);
});
