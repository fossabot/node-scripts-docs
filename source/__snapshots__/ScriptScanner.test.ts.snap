// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads existing manifests (monorepo) 1`] = `
Array [
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "monorepo",
    "scriptCode": "tsc",
    "scriptName": "build",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "tsc",
    "scriptName": "build",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material build --site-dir=public",
    "scriptName": "docs:build",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "yarn nsd-debug --include-locals",
    "scriptName": "docs:scripts",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material",
    "scriptName": "docs:serve",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "yarn build && npm publish",
    "scriptName": "npm:publish",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "yarn build && npm version patch && npm publish",
    "scriptName": "npm:publish:patch",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "yarn build && node output/node-scripts-docs.js",
    "scriptName": "nsd-debug",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "monorepo",
    "scriptCode": "jest",
    "scriptName": "test",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "jest",
    "scriptName": "test",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "packages/some-project/package.json",
    "projectName": "some-project",
    "scriptCode": "jest --coverage",
    "scriptName": "test:coverage",
  },
]
`;

exports[`loads existing manifests (simple project) 1`] = `
Array [
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "tsc",
    "scriptName": "build",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material build --site-dir=public",
    "scriptName": "docs:build",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "yarn nsd-debug --include-locals",
    "scriptName": "docs:scripts",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material",
    "scriptName": "docs:serve",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "yarn build && npm publish",
    "scriptName": "npm:publish",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "yarn build && npm version patch && npm publish",
    "scriptName": "npm:publish:patch",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "yarn build && node output/node-scripts-docs.js",
    "scriptName": "nsd-debug",
  },
  ScriptStoreEntry {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "jest",
    "scriptName": "test",
  },
  ScriptStoreEntry {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "default",
    "scriptCode": "jest --coverage",
    "scriptName": "test:coverage",
  },
]
`;
